include/tester/vtcp/VTCPSession.hpp

```diff
@@ -4,55 +4,57 @@
 #include<memory>
 #include<unordered_map>
 
-#include"generic/RequestHandler.hpp"
+#include"tcp/TCPSession.hpp"
 #include"VTCPOpcode.hpp"
 
-class VTCPRequestHandler : public RequestHandler
+class VTCPSession : public TCPSession
 {
 public:
-    VTCPRequestHandler():
-        RequestHandler()
+    VTCPSession(tcp::socket socket, const std::size_t id,
+    std::function<void(const std::size_t, const Message&)> onMessageReceived,
+    std::function<void(const std::size_t)> onDisconnect):
+        TCPSession(std::move(socket), id, onMessageReceived, onDisconnect)
     {
-        _handlers.emplace(VTCPOpcode::VTCP_CONNECT_REQ, std::bind(&VTCPRequestHandler::onVtcpConnect, this, std::placeholders::_1, std::placeholders::_2));
-        _handlers.emplace(VTCPOpcode::VTCP_DISCONNECT_REQ, std::bind(&VTCPRequestHandler::onVtcpDisconnect, this, std::placeholders::_1, std::placeholders::_2));
-        _handlers.emplace(VTCPOpcode::VTCP_AUTO_TEST_REQ, std::bind(&VTCPRequestHandler::onVtcpAutoTest, this, std::placeholders::_1, std::placeholders::_2));
-        _handlers.emplace(VTCPOpcode::VTCP_MANUAL_TEST_REQ, std::bind(&VTCPRequestHandler::onVtcpManualTest, this, std::placeholders::_1, std::placeholders::_2));
+        _handlers.emplace(VTCPOpcode::VTCP_CONNECT_REQ, std::bind(&VTCPSession::onVtcpConnect, this, std::placeholders::_1));
+        _handlers.emplace(VTCPOpcode::VTCP_DISCONNECT_REQ, std::bind(&VTCPSession::onVtcpDisconnect, this, std::placeholders::_1));
+        _handlers.emplace(VTCPOpcode::VTCP_AUTO_TEST_REQ, std::bind(&VTCPSession::onVtcpAutoTest, this, std::placeholders::_1));
+        _handlers.emplace(VTCPOpcode::VTCP_MANUAL_TEST_REQ, std::bind(&VTCPSession::onVtcpManualTest, this, std::placeholders::_1));
     }
 
-    void handle(std::shared_ptr<TCPSession> sender, const Message& request) override
+    void handle(const Message& request) override
     {
         try
         {
             auto opcode = static_cast<VTCPOpcode>(request.readInteger());
-            _handlers.at(opcode)(std::move(sender), request);
+            _handlers.at(opcode)(request);
         }
         catch(const std::exception& e)
         {
             std::cout << e.what() << std::endl;
         }
     }
 
-    ~VTCPRequestHandler() = default;
+    ~VTCPSession() = default;
 
-    void onVtcpConnect(std::shared_ptr<TCPSession> sender, const Message & data)
+    void onVtcpConnect(const Message & data)
     {
         std::cout << "New client connected" << std::endl;
         Message result;
         result.push(static_cast<int>(VTCPOpcode::VTCP_CONNECT_RES));
         std::string result_msg = "Hello new client";
         result.push(result_msg);
         result.pushSize();
-        sender->send(result.getPayload());
+        send(result.getPayload());
```

Comment:
Refactor `TCPSession::send` to take a `const Message &message` as a parameter rather than the payload vector, and use the `getPayload()` in it.

Reply:
done